File: Laba5/C#/Task2/Circle.cs
using static System.Math;
namespace ConsoleApp1
{
    public class Circle : Figure
    {
        public Circle(int r) : base(r)
        {
            this.data = new[] {r};
        }

        protected internal override double calculateArea()
        {
            return PI * Pow(data[0], 2);
        }

        protected internal override double calculateLength()
        {
            return 2 * PI * data[0];
        }
    }
}

File: Laba5/C++/Task2/Circle.cpp
#include "Circle.h"


Circle::Circle(int r) : Figure(&r) {
	this->data = &r;
}


double Circle::CalculateLength() {
	return 2 * M_PI * data[0];
}


double Circle::CalculateArea() {
	return M_PI * pow(data[0], 2);
}